### PR TITLE
Reduce the number of tests for supported catalog formats.

### LINKED ISSUE #1481: devdeps failing with "ModuleNotFoundError: No module named 'pyarrow'"
https://github.com/spacetelescope/romancal/actions/runs/11598260944/job/32293762396#step:10:922

I believe this may be due to the use of a development version of astropy.

This also appears in the downstream/devdeps testing in stcal:
https://github.com/spacetelescope/stcal/actions/runs/11597106821/job/32289791188#step:10:745

Is it necessary/useful to test all of these formats?
https://github.com/spacetelescope/romancal/blob/2f7d0652d17e06207cc6c6263818ab339a1b17db/romancal/tweakreg/tests/test_astrometric_utils.py#L484-L493
It seems like more a test of astropy than anything romancal related.

### PR DESCRIPTION
**Pull request by mairanteodoro**
Closes #1481

This PR reduces the number of supported formats to be tested in the unit test runs. It doesn't mean we're reducing the number of actual supported formats, which should be the same as those supported by `astropy`'s `table.Table` class (full list can be created by running `Table.write.list_formats()`).

### Regression tests
All regtests are passing:
- https://github.com/spacetelescope/RegressionTests/actions/runs/11693029998

## Tasks
- [ ] **request a review from someone specific**, to avoid making the maintainers review every PR
- [x] add a build milestone, i.e. `24Q4_B15` (use the [latest build](https://github.com/spacetelescope/romancal/milestones) if not sure)
- [ ] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [x] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [x] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/romancal/wiki/How-to-resolve-JIRA-issues)

<details><summary>news fragment change types...</summary>

  - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
  - ``changes/<PR#>.docs.rst``
  - ``changes/<PR#>.stpipe.rst``
  - ``changes/<PR#>.associations.rst``
  - ``changes/<PR#>.scripts.rst``
  - ``changes/<PR#>.mosaic_pipeline.rst``
  - ``changes/<PR#>.patch_match.rst``

  ## steps
  - ``changes/<PR#>.dq_init.rst``
  - ``changes/<PR#>.saturation.rst``
  - ``changes/<PR#>.refpix.rst``
  - ``changes/<PR#>.linearity.rst``
  - ``changes/<PR#>.dark_current.rst``
  - ``changes/<PR#>.jump_detection.rst``
  - ``changes/<PR#>.ramp_fitting.rst``
  - ``changes/<PR#>.assign_wcs.rst``
  - ``changes/<PR#>.flatfield.rst``
  - ``changes/<PR#>.photom.rst``
  - ``changes/<PR#>.flux.rst``
  - ``changes/<PR#>.source_detection.rst``
  - ``changes/<PR#>.tweakreg.rst``
  - ``changes/<PR#>.skymatch.rst``
  - ``changes/<PR#>.outlier_detection.rst``
  - ``changes/<PR#>.resample.rst``
  - ``changes/<PR#>.source_catalog.rst``
</details>

### CODE REVIEW

File: romancal/tweakreg/tests/test_astrometric_utils.py [hunk 1]
```diff
@@ -474,24 +474,7 @@ def test_create_astrometric_catalog_write_results_to_disk(tmp_path, base_image):
...
-            "asdf",
-            "fits",
-            "hdf5",
-            "parquet",
-            "pandas.html",
-            "pandas.json",
-            "pandas.csv",
-        ]
-    ]
+    list_of_supported_formats = ["ascii.ecsv", "fits"]
```
Comment (braingram) [inline romancal/tweakreg/tests/test_astrometric_utils.py:477]: What would be required to add `asdf`?

**Review by braingram — APPROVED** [review]
LGTM. Fixes the devdeps failure (the 2 remaining failures are the python 3.13 ones which won't run because of scikit-image).

I left a comment about supporting `asdf`. As it wasn't a supported format before I'd say a separate PR makes sense.

**Comment by schlafly** [comment]
FWIW, it looks to me like asdf round-trips fine (tab.write('blah.asdf') ; tab = Table.read('blah.asdf')) seems fine to me.  So I would have guessed that we could include it as a format without additional effort.

**Comment by braingram** [comment]
> FWIW, it looks to me like asdf round-trips fine (tab.write('blah.asdf') ; tab = Table.read('blah.asdf')) seems fine to me. So I would have guessed that we could include it as a format without additional effort.

The `overwrite` argument here causes the `asdf` format to fail:
https://github.com/spacetelescope/romancal/blob/dad64d571e701e2cafa29e61f0ab34fd9aebfa24/romancal/tweakreg/astrometric_utils.py#L153

**Pull request merged**